lxd/backup/backup_instance_config.go

```diff
@@ -112,11 +112,15 @@ func UpdateInstanceConfigStoragePool(c *db.Cluster, b Info) error {
 		return err
 	}
 
-	for _, snap := range b.Snapshots {
-		err = f(shared.VarPath("storage-pools", pool.Name, "containers-snapshots", project.Prefix(b.Project, b.Name), snap, "backup.yaml"))
-		if err != nil {
-			return err
+	// BTRFS snapshots are readonly so cannot update.
+	if pool.Driver != "btrfs" {
```

Comment:
What's that code doing anyway, the snapshots aren't guaranteed to be mounted unless they're started and with the new logic, they're all going to be read-only anyway once mounted.

We shouldn't be messing with `backup.yaml` inside snapshots. Instead we should make sure to re-generate `backup.yaml` as soon as a snapshot is restored so that the file in the container volume is always correct.

Reply:
@stgraber as discussed on IRC, this has uncovered an existing bug that occurs when a backup is restored to a different storage pool. This is happening every time now due to a simplification I added to always rewrite the storage pool in the backup.yaml file rather than only if the storage pool has changed. As you suggested, I'll modify this so snapshot's backup.yaml is only re-written on snapshot restore, not on backup restore. Thanks